pkg/middlewares/auth/forward.go

```diff
@@ -89,7 +90,7 @@ func NewForward(ctx context.Context, next http.Handler, config dynamic.ForwardAu
 		fa.authResponseHeadersRegex = re
 	}
 
-	return fa, nil
+	return connectionheader.Remove(fa), nil
```

Comment:
This call does not remove anything, so I think it is misleading. I'd call the method something like "Remover" instead, as it returns something that is in charge of removing.